### ISSUE TITLE
[BUG] dnFBDP does not accept "Phi" as an alias of "rho"

### ISSUE THREAD

**Issue by davidcerny** [revbayes/revbayes#456]
**Describe the bug**
Apparently these two functions will never cease giving us grief...  As I was experimenting with the [mass extinction tutorial](https://revbayes.github.io/tutorials/divrate/efbdp_me.html) for PR #455, I noticed a few issues that likely stem from the recent refactoring of `dnBDSTP` and `dnFBDP` (PR #440):

- In `dnFBDP`, we lost the ability to use `Phi` as an alias for `rho`.
- Unlike `dnFBDP`, which doesn't care, `dnBDSTP` is sensitive to whether its arguments are of type `Probability` or `RealPos`. Attempting to replace the `dnFBDP` call in the tutorial scripts with a `dnBDSTP` will result in the latter function complaining about how `Phi` is of type `RealPos` when it should be of type `Probability` or `Probability[]`.
- When this problem is fixed by explicit coercion (`sampling_at_present <- Probability(14/23)`), I still get an error, despite the fact there is no contradiction between the provided and expected calls:

```
> source("/Users/David/Downloads/mcmc_CRFBD.Rev")
   Processing file "/Users/David/Downloads/mcmc_CRFBD.Rev"
   Attempting to read the contents of file "crocs_T1.tre"
   Successfully read file
   Error:	Argument or label mismatch for function call.
   Provided call:
   dnBDSTP (RealPos<deterministic> 'rootAge',
            RealPos<stochastic> 'lambda',
            RealPos<stochastic> 'mu',
            RealPos<stochastic> 'phi',
            Probability<constant> 'Phi',
            String<constant> 'condition',
            Taxon[]<constant> 'taxa',
            TimeTree<constant> 'initialTree' )
   
   Correct usage is:
   dnBirthDeathSamplingTreatment (RealPos<any> rootAge,
                                  RealPos|RealPos[]<any> lambda,
                                  RealPos|RealPos[]<any> mu,
                                  RealPos|RealPos[]<any> phi,
                                  RealPos[]<any> timeline,
                                  RealPos[]<any> lambdaTimeline,
                                  RealPos[]<any> muTimeline,
                                  RealPos[]<any> phiTimeline,
                                  RealPos[]<any> PhiTimeline,
                                  String<any> condition {valid options: "time"|"survival"|"sampling"},
                                  Taxon[]<any> taxa,
                                  TimeTree<any> initialTree,
                                  Probability[]<any> Lambda,
                                  Probability[]<any> Mu,
                                  RealPos[]<any> LambdaTimeline,
                                  RealPos[]<any> MuTimeline,
                                  Probability|Probability[]<any> Phi,
                                  Probability[]<any> R,
                                  Probability|Probability[]<any> r,
                                  RealPos[]<any> rTimeline)
   
   Error:	Problem processing line 52 in file "/Users/David/Downloads/mcmc_CRFBD.Rev"
```

**To reproduce**
Run scripts `mcmc_CRFBD.Rev` or `mcmc_EFBD_mass_extinctions.Rev` from the [mass extinction tutorial](https://revbayes.github.io/tutorials/divrate/efbdp_me.html).

**Expected behavior**
`dnFBDP` should accept `Phi` as an alias for `rho`; `dnBDSTP` should be able to coerce `RealPos` arguments to type `Probability`.

**Computer info**
macOS 14.4.1 (Sonoma), RevBayes 1.2.2, current version of `development` (rapture-2545-g7c2b56).

**Comment by bjoelle** [comment]
so the Phi thing is not due to my refactoring - dnFBDP never accepted Phi as an argument and I kept that behaviour when I made my changes
I don't think I made any changes to argument types either but I can look into it

**Comment by bjoelle** [comment]
I can't reproduce the second issue - script works fine for me (Ubuntu 20.04, same version of RevBayes)
I think your call to dnBDSTP is missing the r argument ? following my changes there is no longer a default on that one

**Comment by davidcerny** [comment]
Oops, I'm sorry – both for implying that it was your recent PR that gave rise to these issues @bjoelle, and for completely forgetting about our newly implemented requirement to specify `r` when calling `dnBDSTP`. The latter is indeed sufficient to fix the provided/expected call mismatch.

I've renamed the issue to make it less misleading. I still think `dnFBDP` must have accepted `Phi` as an alias of `rho` at some point in time; otherwise the code given in the mass extinction tutorial would have never worked. But this alone should be a trivial thing to fix (I might incorporate it into #455, which requires more changes anyway).

**Comment by hoehna** [comment]
Yes, I think there was a brief time where `dnBDSTP` and `dnFBDP` were aliases.

**Comment by davidcerny** [comment]
Fixed by PR #455.
